src/algorithm/frames.hpp

```diff
@@ -263,6 +262,87 @@ namespace pinocchio
                                      const ReferenceFrame rf,
                                      const Eigen::MatrixBase<Matrix6xLike> & dJ);
 
+/**
+  * @brief Get the inertia supported by a specific frame (given by frame_id), inside of the frame rigid body, expressed in the LOCAL frame.
+  *        The supported inertia in body corresponds to the sum of :
+  *         * The frame inertia
+  *         * The 'child frames' inertia. ('Child frames' refers to frames that share the same parent joint and are placed after the given frame)
+  *
+  * @note Physically speaking, if the body contaning the current frame were to be cut in two parts at that frame, the FrameSupportedInertiaInBody would be the inertia of the part of the body that was after the frame.
+  *
+  * @note The equivalent function for a joint would be to read `model.inertia[joint_id]`.
+  *
+  * @tparam JointCollection Collection of Joint types.
+  *
+  * @param[in] model The model structure of the rigid body system.
+  * @param[in] frameId The index of the frame.
+  *
+  * @return The computed inertia.
+  */
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+  InertiaTpl<Scalar, Options>
+  computeFrameSupportedInertiaInBody(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+                                     const FrameIndex frame_id);
+
+/**
+  * @brief Get the total inertia supported by a specific frame (given by frame_id) expressed in the LOCAL frame.
+  *        The total supported inertia corresponds to the sum of all the inertia after the given frame, i.e :
+  *         * The frame supported inertia in body (see computeFrameSupportedInertiaInBody)
+  *         * The child joints inertia
+  *        You must first call pinocchio::forwardKinematics to update placement values in data structure.
+  *
+  * @note Physically speaking, if the robot were to be cut in two parts at that given frame, this supported inertia would represents the inertia of the part that was after the frame.
+  *
+  * @note The equivalent function for a joint would be to read `data.Ycrb[joint_id]`, after having called pinocchio::crba.
+  *
+  * @tparam JointCollection Collection of Joint types.
+  *
+  * @param[in] model The model structure of the rigid body system.
+  * @param[in] data The data structure of the rigid body system.
+  * @param[in] frameId The index of the frame.
+  *
+  * @return The computed inertia.
+  *
+  * @warning forwardKinematics should have been called first
+  */
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+  InertiaTpl<Scalar, Options>
+  computeFrameSupportedInertiaWithSubtree(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
```

Comment:
maybe using the naming `computeInertiaSupportedByFrame` is even more explicit.
I think merging the two functions in a single one would be much more coherent.
We just need to add an optional argument `bool with_subtree=false`